### PR TITLE
[NASA/NOS3#93] Sample Component

### PR DESCRIPTION
**Pull request by jlucas9**
Complete overhaul of the sample component including:
* cFS Application
* GSW Database
* NOS3 Simulation
* Standalone Checkout
* README.md

Proven working on [nos3#93-SampleUpdate e5de02](https://github.com/nasa/nos3/commit/e5de021736390648f7ce639a1c6e894b223d4791) branch.

### CODE REVIEW

File: fsw/src/sample_app.c [hunk 1]
```diff
@@ -1,461 +1,608 @@
...
+
+    /* Check that device is enabled */
+    if (SAMPLE_AppData.HkTelemetryPkt.DeviceEnabled == SAMPLE_DEVICE_ENABLED)
+    {
+        /* Open device specific protocols */
+        status = uart_close_port(SAMPLE_AppData.SampleUart.handle);
+        if (status == OS_SUCCESS)
+        {
+            SAMPLE_AppData.HkTelemetryPkt.DeviceCount++;
+            SAMPLE_AppData.HkTelemetryPkt.DeviceEnabled = SAMPLE_DEVICE_ENABLED;
```
Comment (jlucas9) [inline fsw/src/sample_app.c]: Copy paste issue identified by @msuder - Should be `SAMPLE_DEVICE_DISABLED`. May have a larger issue with NOS Engine reconnecting when enable / disable cycled.

Comment (jlucas9) [reply fsw/src/sample_app.c]:
```
EVS Port1 42/1/SAMPLE 16: SAMPLE: Disable command received
2022-12-14 19:02:02.921647 [ ERROR ] NosEngine.Common - receive operator failed to receive message, connection was shutdown in NosEngine::Transport::Connection::process_stop(std::unique_lock<std::mutex> &) at transport/src/Connection.cpp:377
2022-12-14 19:02:02.921813 [ ERROR ] NosEngine.Common - receive operator caught exception while receiving message, will stop receiving, message connection was shutdown, before async receive message started
EVS Port1 42/1/SAMPLE 17: SAMPLE: Device disabled
```

Comment (jlucas9) [reply fsw/src/sample_app.c]: Resolved in commit https://github.com/nasa-itc/sample/pull/1/commits/0474cb32ec8a8612c5c09ae6ddf97b652a7b98f4

File: README.md [hunk 2]
```diff
@@ -1,23 +1,112 @@
...
+* uint8, command identifier
+  - (0) Get Housekeeping
+  - (1) Get Sample
+  - (2) Set Configuration
+* uint32, command payload
+  - Unused for all but set configuration command
+* uint16, 0xBEEF
+
+## Telemetry
+Telemetry formats are as follows:
```
Comment (jlucas9) [inline README.md]: https://github.com/nasa/nos3/pull/109#issuecomment-1353405028
So from the physical hardware accepts commands and produces responses. 
The FSW accepts commands and reports telemetry.
@msuder - Is that what you're thinking the terminology should be updated to reflect?

Comment (msuder) [reply README.md]: Yes

**Review by Isgaroth-the-Green — COMMENTED** [review]
This all looks reasonable to me (with the caveat that I am likely the team member least knowledgeable about what would look good or bad).

File: fsw/mission_inc/sample_platform_cfg.h [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/************************************************************************
```
Comment (jlucas9) [inline fsw/mission_inc/sample_platform_cfg.h]: @mgrubb-gsfc  - This file should be in the `platform_inc` directory

Comment (jlucas9) [reply fsw/mission_inc/sample_platform_cfg.h]: Resolved in commit https://github.com/nasa-itc/sample/pull/1/commits/78c4887258bf031e6dd325db3b81f0502ad82d7f

File: README.md (hunk 2 above)
Comment (jlucas9) [reply README.md]: Resolved in commit 31ce518

**Pull request merged**